infra/modules/azure_app_service/locals.tf

```diff
@@ -15,9 +20,9 @@ locals {
   }
 
   app_service = {
-    sku_name               = var.tier == "test" ? "B1" : var.tier == "standard" ? "P0v3" : "P1v3"
-    zone_balancing_enabled = var.tier != "test"
-    is_slot_enabled        = var.tier == "test" ? 0 : 1
+    sku_name               = local.tier == "s" ? "B1" : local.tier == "m" ? "P0v3" : local.tier == "l" ? "P1v3" : local.tier == "xl" ? "P2v3" : "B1"
+    zone_balancing_enabled = local.tier != "s"
+    is_slot_enabled        = local.tier == "s" ? 0 : 1
```

Comment:
```suggestion
sku_name               = lookup({
      "xs" = "B1"
       "s"  = "B1"
      "m"  = "P0v3"
       "l"  = "P1v3"
        "xl" = "P2v3"
       }, local.tier, "B1")  
    zone_balancing_enabled = local.tier != "s"
    is_slot_enabled        = local.tier == "s" ? 0 : 1
```

Reply:
even better: create a centralized `tiers.tf `file to define all tier-related variables and mappings and use them inside every locals.tf ie:

`sku_name               = local.sku_name_mapping[local.tier]`

with `tiers.tf`

```
+/**
+ * Tier Definitions
+ * Centralized mapping for legacy and new tiers
+ */
+
+locals {
+  # Tier mappings for better scalability and readability
+  tier_mapping = {
+    "test"     = "s"
+    "standard" = "m"
+    "premium"  = "l"
+    "xs"       = "xs"
+    "s"        = "s"
+    "m"        = "m"
+    "l"        = "l"
+    "xl"       = "xl"
+  }
+
+  # Default to the mapped tier or retain the original tier if not found
+  tier = lookup(local.tier_mapping, var.tier, var.tier)
+
+  # Worker process count based on tier
+  worker_process_count = {
+    "xs" = 1
+    "s"  = 2
+    "m"  = 4
+    "l"  = 8
+    "xl" = 8
+  }[local.tier]
+
+  # SKU name mapping
+  sku_name_mapping = {
+    "xs" = "B1"
+    "s"  = "B1"
+    "m"  = "P0v3"
+    "l"  = "P1v3"
+    "xl" = "P2v3"
+  }
+}

```